spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -565,10 +561,88 @@ def _measure_tissue_bridges(self, im_lesion_data, p_lst, idx):
     def _measure_length(self, im_data, p_lst, idx):
         """
         Measure the length of the lesion along the superior-inferior axis when taking into account the angle correction
+        The length is computed across all sagittal lesion (i.e., midsagittal and parasagittal) slices meaning that the
+        measurement is 3D.
+        For lesion length for the midsagittal slice only, see _measure_length_midsagittal_slice().
         """
         length_cur = np.sum([p_lst[2] / np.cos(self.angles_3d[zz]) for zz in np.unique(np.where(im_data)[2])])
         self.measure_pd.loc[idx, 'length [mm]'] = length_cur
-        printv('  (S-I) length : ' + str(np.round(length_cur, 2)) + ' mm', self.verbose, type='info')
+        printv('  (S-I) length: ' + str(np.round(length_cur, 2)) + ' mm', self.verbose, type='info')
+
+    def _measure_length_midsagittal_slice(self, im_lesion_data, p_lst, idx):
+        """
+        Measure the length of the lesion along the superior-inferior axis in the **midsagittal slice** when taking into
+        account the angle correction.
+
+        :param im_lesion_data: 3D numpy array: mask of the lesion. The orientation is assumed to be RPI (because we
+        reoriented the image to RPI using orient2rpi())
+        :param p_lst: list, pixel size of the lesion
+        :param idx: int, index of the lesion
+        """
+        # Fetch a list of axial slice numbers that are nonzero in the mid-sagittal slice (RPI)
+        im_data_midsagittal = im_lesion_data[self.midsagittal_sc_slice_rpi, :, :]  # 3D -> 2D, dim=[AP, SI]
+        nonzero_axial_slices = np.unique(np.where(im_data_midsagittal)[1])  # [1] -> SI
+
+        # Check if the lesion exists in the midsagittal slice, if not, set the length to 'np.nan' and return
+        if not np.any(im_data_midsagittal):
+            length_cur = np.nan
+        else:
+            # Compute the length of the lesion along the superior-inferior axis in the midsagittal slice
+            # The length is computed as the sum of the angle corrected axial slice thicknesses
+            length_cur = np.sum([p_lst[2] / np.cos(self.angles_sagittal[s])  # p_lst[2] -> pixel size of SI axis
+                                 for s in nonzero_axial_slices])
```

Comment:
Thank you for thinking about the case when there are no nonzero axial slices! 

FWIW, [`np.sum([]) = 0.0`](https://numpy.org/doc/stable/reference/generated/numpy.sum.html), so there wouldn't be an error? But, if we prefer `np.nan`, then this is good! :)

Reply:
Yeah, I believe `np.nan` is better because it explicitly indicates that the particular lesion (in the case of multiple lesions) is not captured on the midsagittal slice. `0.0.` could indicate that the lesion is presented but has length == 0.

Reply:
Agreed!